x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -30,27 +24,25 @@ public final class LogicalVerifier {
     private LogicalVerifier() {}
 
     /** Verifies the optimized logical plan. */
-    public Collection<Failure> verify(LogicalPlan plan) {
-        Set<Failure> failures = new LinkedHashSet<>();
+    public Failures verify(LogicalPlan plan) {
+        Set<Failure> set = new LinkedHashSet<>();
+        Failures f = new Failures(set);
```

Comment:
Nit: I think shortening the variable name to a single letter here decreases the readability.